backend/src/common/embedding-provider/index.ts

```
@@ -0,0 +1,49 @@
+export interface ModelProviderConfig {
+    endpoint: string;
+    defaultModel?: string;
+  }
+
+export class EmbeddingProvider {
+    private static instance: EmbeddingProvider | undefined = undefined;
+    private logger = new Logger(EmbeddingProvider.name);
+    constructor(private readonly httpService: HttpService, private readonly config: ModelProviderConfig) {}
+    
+    public static getInstance() {
+        if(!this.instance){
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)